### ISSUE TITLE
Trial never passes a traceback to pyunit-API TestResult objects

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2495]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2495|
|Type|defect|
|Created|2007-02-28 07:24:57Z|

If one is using the `pyunit.TestResult` object, or the zope testrunnner `TestResult` object, one never, ever receives a traceback from Trial `TestCase`s.

Extremely frustrating bug.

<details><summary>Searchable metadata</summary>

```
trac-id__2495 2495
type__defect defect
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1172647497000000 1172647497000000
changetime__1173170561000000 1173170561000000
version__None None
owner__ 
cc__itamarst
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml set owner to spiv|
|-|-|

Ready for review in pyresult-tb-2495.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_2"></a>|@jml set owner to @jml|
|-|-|

Taking down from review and adding traceback simulation feature.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_3"></a>|@jml set owner to spiv|
|-|-|

Ready for review in same branch.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

What is the relation of this branch re the fixes proposed for [#2466](https://github.com/twisted/twisted/issues/2466)?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml commented|
|-|-|

This branch makes no changes to the handling of locals or globals. The only new attribute it introduces is a method.

Changing the handling of locals or globals probably wouldn't affect the branch.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @jml|
|-|-|

FormattableTraceback, Code, and Frame are missing docstrings and attribute documentation.  I'm not sure **Formattable**Traceback makes a lot of sense (if it does, why not **Formattable**Code and **Formattable**Frame?)

```getFormattableTraceback```'s docstring is misformated.  I'm also not sure what it's behavior in the case of Failure(Exception("foo")) is meant to be.

In ```test_getFormattableTraceback```, ```assertEqual``` should be ```assertIdentical```, shouldn't it be?

TestFormattableTraceback is missing docstrings.

```ErrorTest``` defined in ```test_traceback``` gratuitously uses ```s``` instead of ```self```.

Also, should the todo on ```test_traceback``` be there?  The test passes, and so does the referenced precondition.

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_7"></a>|@jml set status to `closed`|
|-|-|

(In [19771]) Useful tracebacks when running Trial tests against pyunit `TestResult`s

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#2495](https://github.com/twisted/twisted/issues/2495)

Adds a new method `getTracebackObject` to `Failure`, which always returns 
an object that can be passed to `traceback.extract_tb` and includes all the
information the `Failure` has on the relevant stack trace.

Uses this new method to provide `sys.exc_info`-like information to pyunit
`TestResult` objects via `PyUnitResultAdapter`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
